airbyte-integrations/connectors/source-chargebee/unit_tests/integration/test_addon.py

```diff
@@ -186,16 +186,21 @@ def test_given_no_initial_state_when_read_then_return_state_based_on_most_recent
             _a_response().with_record(_a_record().with_cursor(cursor_value)).build()
         )
         output = self._read(_config().with_start_date(self._start_date - timedelta(hours=8)), _NO_STATE)
-        assert output.most_recent_state == { _STREAM_NAME: {_CURSOR_FIELD: str(self._now_in_seconds) }}
+        most_recent_state = output.most_recent_state
+        assert most_recent_state.stream_descriptor == StreamDescriptor(name=_STREAM_NAME)
+        assert most_recent_state.stream_state == AirbyteStateBlob(updated_at=cursor_value)
 
     @HttpMocker()
     def test_given_initial_state_use_state_for_query_params(self, http_mocker: HttpMocker) -> None:
         # Tests updating query param with state
         state_cursor_value = int((self._now - timedelta(days=5)).timestamp())
+        record_cursor_value = self._now_in_seconds - 1
         state =  StateBuilder().with_stream_state(_STREAM_NAME, {_CURSOR_FIELD: state_cursor_value}).build()
         http_mocker.get(
             _a_request().with_sort_by_asc(_CURSOR_FIELD).with_include_deleted(True).with_updated_at_btw([state_cursor_value, self._now_in_seconds]).build(),
-            _a_response().with_record(_a_record().with_cursor(self._now_in_seconds - 1)).build(),
+            _a_response().with_record(_a_record().with_cursor(record_cursor_value)).build(),
         )
         output = self._read(_config().with_start_date(self._start_date - timedelta(hours=8)), state)
-        assert output.most_recent_state == { _STREAM_NAME: {_CURSOR_FIELD: str(self._now_in_seconds) }}
\ No newline at end of file
+        most_recent_state = output.most_recent_state
+        assert most_recent_state.stream_descriptor == StreamDescriptor(name=_STREAM_NAME)
+        assert most_recent_state.stream_state == AirbyteStateBlob(updated_at=record_cursor_value)
```

Comment:
The change to the test makes sense since now we set the cursor value to the the timestamp of the latest record.

Not a blocking comment to approval, but I'm curious about the old implementation. Before your changem, did this test previously not pass and that's why you had to fix the value? Because the old value `str(self._now_in_seconds)` doesn't the new test value `self._now_in_seconds - 1`

Reply:
I believe this was a change as a result of https://github.com/airbytehq/airbyte-internal-issues/issues/6314

Previously state was being set as the end of the last slice (or "now" in the context of Chargebee) but now it's set as the most recent record.

cc : @girarda

Reply:
right so the change makes sense and we're in a good state now, but I was more curious if in the previous version of the CDK, the slice boundary just happened to be 1 second later than the record which would explain the change to the test. The updated test behavior looks right, I was mainly just curious about the previous state of the test